src/sparsebase/io/edge_list_writer.cc

```diff
@@ -0,0 +1,189 @@
+#include "sparsebase/io/edge_list_writer.h"
+
+#include <string>
+
+#include "sparsebase/config.h"
+#include "sparsebase/io/sparse_file_format.h"
+#include "sparsebase/io/writer.h"
+
+namespace sparsebase::io {
+
+template <typename IDType, typename NNZType, typename ValueType>
+EdgeListWriter<IDType, NNZType, ValueType>::EdgeListWriter(
+    std::string filename, bool directed)
+    : filename_(filename),
+      directed_(directed) {}
+
+template <typename IDType, typename NNZType, typename ValueType>
+void EdgeListWriter<IDType, NNZType, ValueType>::WriteCOO(
+    format::COO<IDType, NNZType, ValueType> *coo) const {
+  std::ofstream edgeListFile;
+  edgeListFile.open(filename_);
+  if constexpr (std::is_same_v<ValueType, void>) {
+    IDType* row = coo->get_row();
+    IDType* col = coo->get_col();
+    std::vector<std::tuple<IDType, IDType>> edges;
+    for (int i = 0; i < coo->get_num_nnz(); ++i) {
+      IDType u = row[i], v = col[i];
+      if (!this->directed_ && u > v) {
+        std::swap(u, v);
+      }
+      edges.push_back(std::tuple<IDType, IDType>(u, v));
+    }
+    sort(edges.begin(), edges.end(),
```

Comment:
On the reader, this code ensures that there are no duplicates in the COO. For the writer, we can probably assume, there won't be any duplicates and the COO is sorted as that is the default behavior of the library. So the sort, the unique operator, and the erase could possibly be removed. A better approach is adding a parameter that can toggle them, like `remove_duplicates`.

Reply:
Same for all the other sort, unique, erase codes